backend/app/routers/user_profile_route.py

```diff
@@ -321,3 +323,193 @@ def delete_user_profile(
     db_session.delete(user_profile)
     db_session.commit()
     return {'message': 'User profile deleted successfully'}
+
+
+@router.get('/export', response_model=dict)
+def export_user_data(
+    user_profile: Annotated[UserProfile, Depends(require_user)],
+    db_session: Annotated[DBSession, Depends(get_db_session)],
+) -> dict:
+    """
+    Export all user-related data (history) for the currently authenticated user.
+    """
+    user_id = user_profile.id
+
+    # User profile
+    user_data = {
```

Comment:
the file gets a little too large when defining the JSON inside the route.
It would be better to put it inside the models/schemas (whatever fits better)